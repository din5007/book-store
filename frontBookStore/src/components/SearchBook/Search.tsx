import { FC } from "react";
import { Form } from "react-bootstrap";

const SearchBook: FC<{ onChange: any }> = ({ onChange }) => {

  const onChangeHandler = (e: any) => {
    e.preventDefault();
    if(e?.target?.value?.length > 2) {
        onChange(e.target.value);
    }
  }

  return (
    <Form.Group className="mb-3" controlId="exampleForm.ControlInput1">
      <Form.Control type="search" placeholder="Search by Book Name" onChange={onChangeHandler}/>
    </Form.Group>
  );
};

export default SearchBook;
