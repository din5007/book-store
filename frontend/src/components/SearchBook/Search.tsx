import { EventHandler, FC, KeyboardEventHandler, SyntheticEvent } from "react";
import { Form } from "react-bootstrap";

const SearchBook: FC<{ onChange: any }> = ({ onChange }) => {

  const onChangeHandler = (e: any) => {
    e.preventDefault();
    if(e?.target?.value > 3) {
        onChange(e.target.value);
    }
  }

  return (
    <Form.Group className="mb-3" controlId="exampleForm.ControlInput1">
      <Form.Control type="search" placeholder="Search by Book Name" onChange={onChangeHandler}/>
    </Form.Group>
  );
};

export default SearchBook;
