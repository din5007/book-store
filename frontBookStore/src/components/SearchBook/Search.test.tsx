import { render } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from "vitest";
import SearchBook from "./Search";

describe('Search Component', () => {
    it('Search Component should load properly - snapshot', async () => {
        const mock = vi.fn();
        const {container}  = render(<SearchBook onChange={mock}/>);
        expect(container).toMatchSnapshot();
    });

    it('Search Component should load properly', async () => {
        const mock = vi.fn();
        const {getByPlaceholderText}  = render(<SearchBook onChange={mock}/>);
        const input = getByPlaceholderText('Search by Book Name');
        await userEvent.type(input, 'book');
        expect(mock).toBeCalled()
    });
})