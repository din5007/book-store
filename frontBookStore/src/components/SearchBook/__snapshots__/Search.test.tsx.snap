// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Component > Search Component should load properly - snapshot 1`] = `
<div>
  <div
    class="mb-3"
  >
    <input
      class="form-control"
      id="exampleForm.ControlInput1"
      placeholder="Search by Book Name"
      type="search"
    />
  </div>
</div>
`;
